### PR TITLE
Check notebook status by container statuses instead of pod phase

### LINKED ISSUE #507: [Bug]: We redirect before the Pod is Running the Notebook
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

There is an edge case scenario where we will redirect to the Notebook route (eg JL) without it being fully initialized. The user ends up at a "Application is not available" screen.

Leading guess is that we check the Pod for the Running status & not the containers within'. It is possible the OAuth container started, Pod said Running, and we redirected to the Notebook before that container was finished spinning up. 

### Expected Behavior

Never to redirect to the Notebook when it is not running -- we should have better "isNotebookRuning" logic. 

### Steps To Reproduce

It is sporadic, testing over and over, maybe with different images may be able to cause the issue to happen with more consistency.

Creating a notebook is all you need to do to cause this (when it happens).

### Workaround (if any)

_No response_

### OpenShift Infrastructure Version

_No response_

### Openshift Version

_No response_

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

```yml
2.1
```

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
## Description

Closes #507 

## How Has This Been Tested?

1. Create a notebook
2. Watch the ready containers on the cluster for the notebook pod
3. See the success of starting only when there are 2/2 containers ready

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: backend/src/routes/api/notebooks/notebookUtils.ts [hunk 1]
```diff
@@ -52,6 +52,16 @@ export const getNotebook = async (
   }
 };
 
+const checkPodContainersReady = (pod: V1Pod): boolean => {
+  const containerStatuses = pod.status?.containerStatuses || [];
```
Comment (andrewballantyne) [inline backend/src/routes/api/notebooks/notebookUtils.ts:56]:
```suggestion
  const containerStatuses: V1ContainerStatus[] = pod.status?.containerStatuses || [];
```
When you do an `|| []` you mangle the types for TS... I just pulled the type of `containerStatuses` and put it here.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Tested this with my PR #521 and it worked like a charm.

One small change for code quality.

**Review by andrewballantyne — APPROVED** [review]
Looks good, tested it with my PR.

/hold
If someone else wants to verify this PR alone, that'd be great. I'll test later today.

**Review by dlabaj — APPROVED** [review]
/lgtm

**Pull request merged**